### ISSUE TITLE
API issue about deleting old values and player data duplication on database

### ISSUE THREAD

**Issue by xion87** [Zrips/CMI#907]
Hey, my dev is creting the task for allow cmi to change username via api, it work but:
https://gist.github.com/xion87/ad8bf6278ee393edeedc79cbe54aacb8 this code doesn't clear the mysql data while https://gist.github.com/xion87/aad3e32b04f1942ea1361ef3a9415726 if i'm trying to use softremove it returns a nomethodexception as seen

**Comment by Zrips** [comment]
`Caused by: java.lang.NoSuchMethodError: com.Zrips.CMI.PlayerManager.softRemoveUser(Ljava/util/UUID;)V`
Tho removeUser method was not meant to remove player from database itself, like i mentioned before CMI was not made originally to handle things like this in particular, but will add method to remove player from database when using that method. softRemoveUser on another hand removes player from database where you could try to use players object instead of UUID, as it seem to fail to find appropriate method...

**Comment by xion87** [comment]
I tried to do as you say but basically:
playerA log into server and change the name into playerB
playerB log into server 

Here on cmi mysql both rows are saved correct.
I use the custom swap plugin for more the data from playerA to playerB and use your api for delete the old player
Data on db are ok
PlayerB log into server it have all data of old player but.... on mysql? i see 2 rows of same player, CMI is creating a third row and not using the old data.
After the server restart the player lose al progress make in this time (cmi data). The dev think that there is some issue on your api. Since CMI is not recognizing the change and keep to not read the new change make on db

**Comment by LogGits** [comment]
This seems alittle complicated and unnecessary to have in an external plugin. I think this should be resolved in cmi. To me this seems like a fundamental flaw in the original code. 

I think this should be resolved in cmi because this issue will affect:
- offline servers
- online servers (when a player changes their username on the mojang website)
-bungeecord servers (as it essentially utilises offline mode)

Thats a scary thought but this could explain why users have duplicate entries in my database whrn the join with a new name.

**Comment by xion87** [comment]
like now it find 10400 duplicates at each server start that is not true, did you manage to fix this? is pretty critical and is stupid to see this in a premium plugin @Zrips

**Comment by Zrips** [comment]
Will have a look at it, but it can be due to some actions, like scanning offline user files which could cause some duplicate creations. Will have to investigate it.

**Comment by xion87** [comment]
i have like 46000 duplicates on kitpvp and 46 on creative, you need some database dump to check these?

**Comment by LogGits** [comment]
Heya, i have my suspicions on this. I also have this problem. What i think is happening is CMI is for some reason storing 2 versions of each account. The first is the actual account. The second is a username in lowercase stored as a version of the account. Since the lowercase account doesnt exist, it had no data. @xion87 can u please check if this same thing is happening on your end aswell.

**Comment by xion87** [comment]
Yeah same your thing @LogGits 
sample: https://gist.github.com/xion87/b865bf9a7e15e8fbb00d1a6f8bca36a1

**Comment by Zrips** [comment]
latest updates should address this duplicated user creation issue and should no longer make them. If that still gonna be issue, please reopen ticket or make new one with new information.
And will try to optimize duplicate removal method to be more friendly with removing bunch of them in one go.

**Comment by LogGits** [comment]
@xion87 did the duplicate user creation get fixed with the latest update (`8.2.4.0`)?

**Comment by xion87** [comment]
gonna test right now

**Comment by xion87** [comment]
but the console it keep to spam a anoying error (i through was fixed with the new version) https://gist.github.com/xion87/84543e2e3e81cdd72a531c77ea05d269

**Comment by xion87** [comment]
and yeah still duplicated created with latest version: https://gist.github.com/xion87/1d858d7f61fe71ea05e710305afc7d9a

**Comment by xion87** [comment]
The dupe bug is still on 8.2.6.0

**Comment by Zrips** [comment]
@xion87 Did you did multiple wipes of duplicates? As there is still issue with that feature which would not remove all of them in one go but would leave couple of them if you have more then 2 records with same name.

**Comment by xion87** [comment]
i mean:
- the dupe bug is absent in factions for any reason, also i converted into cmi economy and seem to "work" 
- the api for change the user name is still broken and still cause this https://github.com/Zrips/CMI/issues/907#issuecomment-428242956

**Comment by xion87** [comment]
About for this issue, now is pretty critical to have it fixed since i use money and homes on player and need to move data: @Zrips  Can you add the cmi option "save on logout"? Like the jobs plugin, i know is a little lag but i think it will fix my issue

**Comment by xion87** [comment]
any news about this?

**Comment by xion87** [comment]
using the external plugin it alwaw wipe the cmi data of the player

**Comment by LogGits** [comment]
Just to clarify. I appear to still be getting duplicate database entries regardless of the many changes we've made. I manually deleted all duplicate entries from the mysql database. And i recently saw that i now had 4 duplicate entries again. Meaning, the root cause of this issue is still present. To be sure, @xion87 your issue is also regarding duplicate entries in the database and fixing it will resolve issue #907?
